### DISCUSSION TITLE
6812 LEDs not displaying red

### DISCUSSION THREAD

**Discussion by marceskenazi** [awawa-dev/HyperSerialEsp8266#34] [Q&A]
After discussions about HyperHDR v19 liaising with WLED, I decided to try HyperSerial ESP8266 as my ESP 32 doesn't have a fast enough serial chip onboard. My 8266 does have the CH340G chip. It is flashed with firmware_SK6812_RGBW_NEUTRAL V8

I'm powering 230 6812 LEDs and a Pi 3b+ with a 20amp 5v supply. The power for the 8266 is fed by the USB cable from the Pi. I then wired the power for a 74AHCT125 level shifter from the VIN pin on the 8266. 

I can't get the level shifter to work. So as a temporary workaround, I'm taking the data lead for the LEDs from the 8266 D4 pin.

To get anywhere near true colours, I had to use RBG in the adalight controller. All colours are good except for red. The best I can get is a yellow when there should be red. The live video visualization shows red leds when there is a red image.  

Has anyone else come across this issue? Would logs help? I don't think it's an issue with HyperHDR though. 

Many thanks for any suggestions.

**Comment by awawa-dev** [comment]
There is probably some problem with data transmission or you have unusual version of sk6812LEDs, because normally you would not need to change the RGB system. You can only change the order if you are using the sk6812 cold version firmware, the neutral version (which is not neutral at all) expects strictly RGB to calibrate the white channel (or try to manually set it to 160 for R/G/B in the adalight driver). BTW you can try compiling on Github HyperSerialESP32 for a lower speed of 921600, it only takes a few minutes.

**Reply by marceskenazi** [reply]
Once again, thanks for your speedy reply.

The 6812 LEDs are the same ones I've been using for years and have worked well and true with HyperHDR on that same Pi, but using the GPIO pin of the Pi for data, and also with WLED, so I don't think it's the strip.

I'm not sure what could cause a problem with the data transmission. I'll double check the wiring. And I'll re-flash the 8266 with the cold version to see if that helps.

I wish I could try your suggestion of compiling HyperSerialESP32, but I don't have a clue how to do that!

**Reply by awawa-dev** [reply]
HyperSerialESP32 using own algorithms for RGBW calibraton. Before flashing new firmware change R/G/B in custom white channel calibration (HyperHDR Adalight driver section). Set them all to 160 or 0 to see if there is a difference. Also enable 'continues output' in the 'smoothing' and test the LEDs using Color effect in the 'remote' tab.

**Reply by marceskenazi** [reply]
Unfortunately, changing those values had no effect. 

But, having found your excellent instructions on how to compile, I have managed to create a new bin file with the slower speed. I'll flash this to my ESP32 tomorrow and give it a try.

Always glad to learn something new. Thanks again.
